### PR TITLE
test: add error descriptions / narration to kvstore test

### PR DESCRIPTION
**Pull request by ianvernon**
This gives more context when the test fails.

Signed-off by: Ian Vernon <ian@cilium.io>

I did this when I noticed during #5773 that the test failure logs were not conducive for debugging.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5774)

### CODE REVIEW

**Comment by ianvernon** [comment]
Holding off testing until #5775 is merged.

**Review by eloycoto — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

**Pull request merged**
